File: src/main/java/co/com/fsighttech/imp/issuing/bo/core/Policy.java
/**
 * 
 */
package co.com.fsighttech.imp.issuing.bo.core;

import java.time.LocalDate;
import java.util.List;

import lombok.Getter;
import lombok.Setter;

/**
 * this object represents a policy
 * 
 * @author jtorres
 *
 */
@Getter
@Setter
public class Policy {

	private Company company;
	private Product product;
	private String policyNumber;
	private LocalDate issueDate;
	private LocalDate effectiveDate;
	private LocalDate expiredDate;
	private LocalDate processDate;
	private Long currencyCode;
	private Premium	premium;
	private Insured	insured;
	private Agent	agent;
	private Commision commision;
	private Office	office;
	private Long collectorCode;
	private String collectorType;
	private AdditionaInterest additionalInterest;
	private List<Risk> risks;
	private List<CoverageGroup> coveragesGroups;
	private List<VariableData> variableDataList;
	private List<VariableData> fixedDataList;
	
}

File: bin/static/src/app/issuing-module/shared/model/core/policy.ts
import { Company } from './company';
import { Product } from './product';
import { Premium } from './premium';
import { Insured } from './insured';
import { Agent } from './agent';
import { Commission } from './commission';
import { Office } from './office';
import { AdditionalInterest } from './additional-Interest';
import { Risk } from './risk';
import { VariableData } from './variable-data';
import { CoverageGroup } from './coverage-group';

/**
* this object represents a policy
*
* @author jtorres
*/
export class Policy {

    company: Company;
    product: Product;
    policyNumber: string;
    issueDate: Date;
    effectiveDate: Date;
    expiredDate: Date;
    processDate: Date;
    currencyCode: number;
    premium: Premium;
    insured: Insured;
    agent: Agent;
    commision: Commission;
    office: Office;
    collectorCode: number;
    collectorType: string;
    additionalInterest: AdditionalInterest;
    coveragesGroups: CoverageGroup[];
    risks: Risk[];
    variableDataList: VariableData[];
    fixedDataList: VariableData[];

}